ocs_ci/deployment/aws.py

```diff
@@ -436,6 +442,414 @@ def destroy_cluster(self, log_level="DEBUG"):
+            rhel_pod_obj.name, constants.OCP4_2_REPO, repo_dst_path
+        )
+        # copy the .pem file for our internal repo on all nodes
+        # including ansible pod
+        # get it from URL
+        mirror_pem_file_path = os.path.join(
+            constants.DATA_DIR,
+            constants.INTERNAL_MIRROR_PEM_FILE
+        )
+        dst = "/etc/pki/ca-trust/source/anchors/"
+        pod.upload(rhel_pod_obj.name, mirror_pem_file_path, dst)
+        # Install scp on pod
+        rhel_pod_obj.install_packages("openssh-clients")
+        # distribute repo file to all RHEL workers
+        hosts = [inst.private_dns_name for node, inst in
+                 self.rhel_worker_list.items()]
+        for host in hosts:
+            disable = "sudo yum-config-manager --disable \*"
+            rhel_pod_obj.exec_cmd_on_node(
+                host, pem_dst_path, disable, user=self.rhel_worker_user
+            )
+            rhel_pod_obj.copy_to_server(
+                host, pem_dst_path, f'{repo_dst_path}/{repo_file}',
+                f'/tmp/{repo_file}', user=self.rhel_worker_user
+            )
+            rhel_pod_obj.exec_cmd_on_node(
+                host, pem_dst_path,
+                f'sudo mv /tmp/{repo_file} {repo_dst_path}',
+                user=self.rhel_worker_user
+            )
+            rhel_pod_obj.copy_to_server(
+                host, pem_dst_path,
+                f'{dst}/{constants.INTERNAL_MIRROR_PEM_FILE}',
+                f'/tmp/{constants.INTERNAL_MIRROR_PEM_FILE}',
+                user=self.rhel_worker_user,
+            )
+            rhel_pod_obj.exec_cmd_on_node(
+                host, pem_dst_path,
+                f'sudo mv /tmp/{constants.INTERNAL_MIRROR_PEM_FILE} {dst}',
+                user=self.rhel_worker_user
+            )
+        # copy kubeconfig to pod
+        kubeconfig = os.path.join(
+            self.cluster_path, config.RUN.get('kubeconfig_location')
+        )
+        pod.upload(rhel_pod_obj.name, kubeconfig, "/")
+        pull_secret_path = os.path.join(
+            constants.TOP_DIR,
+            "data",
+            "pull-secret"
+        )
+        pod.upload(rhel_pod_obj.name, pull_secret_path, "/tmp/")
+        host_file = self.build_ansible_inventory(hosts)
+        pod.upload(rhel_pod_obj.name, host_file, "/")
+        # install pod packages
+        rhel_pod_obj.install_packages(constants.RHEL_POD_PACKAGES)
+        # run ansible
+        openshift_ansible_path = "/usr/share/ansible/openshift-ansible"
+        cmd = (
+            f"ansible-playbook -i /hosts --private-key={pem_dst_path} "
+            f"{openshift_ansible_path}/playbooks/scaleup.yml"
+        )
+
+        rhel_pod_obj.exec_cmd_on_pod(cmd, out_yaml_format=False)
+        self.verify_nodes_added(hosts)
+        # remove rhcos workers
+        self.remove_rhcos_workers()
+
+    def remove_rhcos_workers(self):
+        """
+        After RHEL workers are added remove rhcos workers from the cluster
```

Comment:
we cannot have both rhcos and rhel workers?

Reply:
@vasukulkarni , According to doc https://docs.openshift.com/container-platform/4.2/machine_management/adding-rhel-compute.html, it tells user to remove RHCOS nodes at the end. I am not sure if there is requirement for mixed worker cluster.

@hnallurv , Any idea?

Reply:
Yeah I see in the doc, since its a method call, I guess we can take it up later as well, resolving for now.

Reply:
@shylesh We need to test with mixed OS. We have following tests in cluster expansion (tier 2):

OCS-1236 | In UPI environment with CoreOS based OCS (Worker) nodes, Add RHEL 7 based nodes to the existing woking cluster whose storage capacity is at Near Full (usually 85%, but configurable in ceph) and expand the storage capacity using 'Add Capacity'\
-- | --
OCS-1237 | In UPI enevironment with RHEL 7 based OCS (Worker) nodes, Add CoreOS based nodes to the existing woking cluster whose storage capacity is at Near Full (usually 85%, but configurable in ceph) and expand the storage capacity using 'Add Capacity'

Reply:
https://github.com/red-hat-storage/ocs-ci/issues/1070. should be addressed from this issue.

Reply:
Marking this comment as resolved and it will be taken care in https://github.com/red-hat-storage/ocs-ci/issues/1070